standalone-metastore/metastore-tools/tools-common/src/main/java/org/apache/hadoop/hive/metastore/tools/HMSClient.java

```diff
@@ -345,21 +348,44 @@ boolean openTxn(int numTxns) throws TException {
     return openTxns;
   }
 
+  List<TxnInfo> getOpenTxnsInfo() throws TException {
+    return client.get_open_txns_info().getOpen_txns();
+  }
+
   boolean commitTxn(long txnId) throws TException {
     client.commit_txn(new CommitTxnRequest(txnId));
     return true;
   }
 
-  boolean abortTxn(long txnId) throws TException {
-    client.abort_txn(new AbortTxnRequest(txnId));
+  boolean abortTxns(List<Long> txnIds) throws TException {
+    client.abort_txns(new AbortTxnsRequest(txnIds));
     return true;
   }
 
-  boolean abortTxns(List<Long> txnIds) throws TException {
-    client.abort_txns(new AbortTxnsRequest(txnIds));
+  boolean allocateTableWriteIds(String dbName, String tableName, List<Long> openTxns) throws TException {
+    AllocateTableWriteIdsRequest awiRqst = new AllocateTableWriteIdsRequest(dbName, tableName);
+    openTxns.forEach(t -> {
+      awiRqst.addToTxnIds(t);
+    });
+
+    client.allocate_table_write_ids(awiRqst);
     return true;
   }
 
+  boolean getValidWriteIds(List<String> fullTableNames) throws TException {
```

Comment:
I would expect this method to return list of validWriteIds, not just true. Should we change the name?

Reply:
HMClient methods return boolean, so they can be invoked with throwingSupplierWrapper()

Reply:
I don't get what does it have to do with throwingSupplierWrapper. throwingSupplierWrapper just handles checked exceptions.

Reply:
ah sorry, I was mistaken, the reason why it never returned the writeIds is because they are never used, the benchmark is just executing the api call. The return value from the hms is actually a GetValidWriteIdsResponse object, not a list. As it is never used I'm not sure if we need to change this.